src/averageAges.js

```diff
@@ -55,7 +59,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenWithMothersListed = people
+    .filter(child => people.some(mother => mother.name === child.mother)
+      && (!onlyWithSon || child.sex === 'm'))
+    .map(child => ({
```

Comment:
Same here. 
It is better to add some new lines, but to make your code easier to read and understand for other developers.
Shorter doesn't always mean better